backend/inspirehep/pidstore/providers/cnum.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from datetime import datetime
+
+import structlog
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from invenio_pidstore.providers.base import BaseProvider
+
+LOGGER = structlog.getLogger()
+
+
+class CNUMIdentifier(object):
```

Comment:
why a class? this looks like a function to me.

Reply:
Because we need to be consistent. For the control number minter we have a class who does the job and having a class there and a function here makes the code more complicated than having a function instead of a class. And BTW, functions are classes too so I don't see any real improvement :)

Reply:
Sorry, but a class with one method is a function, not a class. If you want to simplify code, change it to a function there. Functions are objects, true, but not classes.

Reply:
Nothing enforces a class to have more than one method :) to make everyone happy I moved that method in the `InspireCNUMProvider` class and removed the other one. I think it fits properly there.